api/prometheus.go

```diff
@@ -249,27 +329,221 @@ func (m Metrics) PrometheusMetric() []prometheus.Metric {
 		},
 		{
-			Name:      "hostd_wallet_confirmed",
-			Value:     wr.Confirmed.Siacoins(),
-			Timestamp: time.Now(),
+			Name: "hostd_wallet_confirmed",
+			Labels: map[string]any{
+				"address": wr.Address,
+			},
+			Value: wr.Confirmed.Siacoins(),
 		},
 		{
-			Name:      "hostd_wallet_unconfirmed",
-			Value:     wr.Unconfirmed.Siacoins(),
-			Timestamp: time.Now(),
+			Name: "hostd_wallet_unconfirmed",
+			Labels: map[string]any{
+				"address": wr.Address,
+			},
+			Value: wr.Unconfirmed.Siacoins(),
 		},
 	}
 }
+
+// PrometheusMetric returns Prometheus samples for the hosts tpool fee.
+func (t TPoolResp) PrometheusMetric() (metrics []prometheus.Metric) {
+	return []prometheus.Metric{
+		{
+			Name:  "hostd_tpool_fee",
+			Value: t.RecommendedFee.Siacoins(),
+		},
+	}
+}
+
+// PrometheusMetric returns Prometheus samples for the hosts volumes.
+func (v VolumeResp) PrometheusMetric() (metrics []prometheus.Metric) {
+	for _, volume := range v {
+		labels := make(map[string]interface{}, 2)
+		labels["id"] = volume.ID
+		labels["local_path"] = volume.LocalPath
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_used_sectors",
+			Labels: labels,
+			Value:  float64(volume.UsedSectors),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_total_sectors",
+			Labels: labels,
+			Value:  float64(volume.TotalSectors),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_read_only",
+			Labels: labels,
+			Value: func() float64 {
+				if volume.ReadOnly {
+					return 1
+				}
+				return 0
+			}(),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_available",
+			Labels: labels,
+			Value: func() float64 {
+				if volume.Available {
+					return 1
+				}
+				return 0
+			}(),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_failed_reads",
+			Labels: labels,
+			Value:  float64(volume.FailedReads),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_failed_writes",
+			Labels: labels,
+			Value:  float64(volume.FailedWrites),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_successful_reads",
+			Labels: labels,
+			Value:  float64(volume.SuccessfulReads),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_successful_writes",
+			Labels: labels,
+			Value:  float64(volume.SuccessfulWrites),
+		})
+		metrics = append(metrics, prometheus.Metric{
+			Name:   "hostd_volume_status",
+			Labels: labels,
+			Value: func() float64 {
+				if volume.Status == "creating" {
+					return 1
+				} else if volume.Status == "ready" {
+					return 2
+				}
+				return 0
+			}(),
+		})
+	}
+	return
+}
+
+// PrometheusMetric returns Prometheus samples for the hosts alerts.
+func (a AlertResp) PrometheusMetric() (metrics []prometheus.Metric) {
+	for _, alert := range a {
+		metrics = append(metrics, prometheus.Metric{
```

Comment:
I think these should include the alert's timestamp so you can show the age of the alert.

Reply:
agreed. will update siagrafana to display with timestamp.